src/api/cvc4cpp.cpp

```diff
@@ -2169,28 +2192,34 @@ DatatypeConstructor Datatype::getConstructorForName(
   }
   CVC4_API_CHECK(foundCons) << "No constructor " << name << " for datatype "
                             << getName() << " exists";
-  return (*d_dtype)[index];
+  return DatatypeConstructor(d_solver, (*d_dtype)[index]);
 }
 
-Datatype::const_iterator::const_iterator(const CVC4::Datatype& dtype,
+Datatype::const_iterator::const_iterator(const Solver* slv,
+                                         const CVC4::Datatype& dtype,
                                          bool begin)
 {
+  d_solver = slv;
```

Comment:
Minor: Could be moved to initializer list.

Reply:
Yeah, didn't move it because the rest is also initialized in the body.